### PR TITLE
Feature/face detection android

### PR DESCRIPTION
**Pull request by pfeiffer**

### CODE REVIEW

**Comment by charpeni** [comment]
Thank you, it seems a nice feature.

I'm concerned about performance and battery usage as this feature is activated by default. I think this should be activated by a boolean prop or only when `onFacesDetected` callback is provided, it will save some unneeded computations and events from `DeviceEventEmitter`.

cc @rt2zz

**Comment by brunolemos** [comment]
+1

File: index.js [hunk 1]
```diff
@@ -142,11 +151,15 @@ export default class Camera extends Component {
...
+    this.cameraFaceDetectionListener = (Platform.OS === 'ios'
+      ? NativeAppEventEmitter
+      : DeviceEventEmitter
+    ).addListener('CameraFaceDetected', this._onFaceDetected);
 
     let { captureMode } = convertNativeProps({captureMode: this.props.captureMode})
     let hasVideoAndAudio = this.props.captureAudio && captureMode === Camera.constants.CaptureMode.video
     let check = hasVideoAndAudio ? Camera.checkDeviceAuthorizationStatus : Camera.checkVideoAuthorizationStatus;
-
+``
```
Comment (sibelius) [inline index.js]: this is a typo

**Comment by sibelius** [comment]
@ptomasroos what do u think of this feature?

**Comment by sibelius** [comment]
@pfeiffer can you rebase with master?

**Comment by sibelius** [comment]
Camera.FaceDetectionListener is deprecated

import android.hardware.Camera; is deprecated as well

**Comment by kinhunt** [comment]
fixed yet?

**Comment by sibelius** [comment]
@kinhunt we this PR https://github.com/react-native-community/react-native-camera/pull/1071

we can do face recognition properly on android using google visio

**Comment by kinhunt** [comment]
@sibelius thank you. #1017 first and then this pr?

**Comment by sibelius** [comment]
you can try RNCamera on master, works on both ios and android

**Pull request closed without merging**